### PR TITLE
Adds everything else OpenDream needs to the sandbox

### PR DESCRIPTION
**Pull request by ike709**
https://github.com/wixoaGit/OpenDream/issues/374

### CODE REVIEW

File: Robust.Shared/ContentPack/Sandbox.yml [hunk 1]
```diff
@@ -13,6 +13,7 @@ AllowedVerifierErrors:
 WhitelistedNamespaces:
 - Robust
 - Content
+- OpenDreamShared
```
Comment (ike709) [inline Robust.Shared/ContentPack/Sandbox.yml:16]: This isn't ideal in the long run but PJB caved so... /shrug

File: Robust.Shared/ContentPack/Sandbox.yml [hunk 2]
```diff
@@ -467,6 +470,8 @@ Types:
     UnmanagedType: { } # `unmanaged` constraint in C# modreqs with this.
   System.Runtime.Versioning:
     TargetFrameworkAttribute: { All: True }
+  System.Text.Json.Serialization:
+    JsonIgnoreAttribute: { All: True }
```
Comment (ike709) [inline Robust.Shared/ContentPack/Sandbox.yml:479]: This is actually just for the ctor, should I change it or does it not really matter?

File: Robust.Shared/ContentPack/Sandbox.yml [hunk 3]
```diff
@@ -299,6 +300,7 @@ Types:
     IEnumerator: { All: True }
     IReadOnlyList`1: { All: True }
   System.ComponentModel:
+    CancelEventArgs: { All: True }
```
Comment (ike709) [inline Robust.Shared/ContentPack/Sandbox.yml:303]: This is actually just for the setter, should I change it or does it not really matter?

File: Robust.Shared/ContentPack/Sandbox.yml (hunk 2 above)
Comment (PJB3005) [reply Robust.Shared/ContentPack/Sandbox.yml:479]: doesn't really matter, though you won't be able to actually make use of these attributes ever anyways as it's reflection-heavy deserialization we can't allow.

**Comment by ike709** [comment]
oops I goofed the whitespace

File: Robust.Shared/ContentPack/Sandbox.yml [hunk 4]
```diff
@@ -332,6 +334,7 @@ Types:
     CompareOptions: { }
     CultureInfo: { All: True }
     DateTimeStyles: { All: True } # Enum
+    NumberStyles: { All: True } # Enum
```
Comment (ike709) [inline Robust.Shared/ContentPack/Sandbox.yml]:
```suggestion
    NumberStyles: { } # Enum
```

**Comment by ike709** [comment]
OpenDream now has everything whitelisted that it needs whitelisted.

**Review by PaulRitter — APPROVED** [review]

File: Robust.Shared/ContentPack/Sandbox.yml [hunk 5]
```diff
@@ -342,14 +345,20 @@ Types:
...
   System.IO:
+    BinaryReader: { All: True }
     FileAccess: { } # Enum
     FileMode: { } # Enum
     FileShare: { } # Enum
     FileNotFoundException: { All: True }
     InvalidDataException: { All: True }
     IOException: { All: True }
     MemoryStream: { All: True }
+    Path: { All: True }
```
Comment (PJB3005) [inline Robust.Shared/ContentPack/Sandbox.yml]: Sorry for looking at this so late but `Path` can be used to probe file system layout (Short path expansion in `Path.GetFullPath()`) and OS setup (is the user on Windows or Linux?).

image

This API should not be exposed to content.

**Review by PJB3005 — CHANGES_REQUESTED** [review]

File: Robust.Shared/ContentPack/Sandbox.yml (hunk 5 above)
Comment (ike709) [inline Robust.Shared/ContentPack/Sandbox.yml]:
```suggestion
```
Path is no longer needed as I moved the entirety of the compiler out of shared.

**Review by PJB3005 — APPROVED** [review]

**Pull request merged**